### PR TITLE
Append Internal paths to Caddy config HiddenFiles

### LINKED ISSUE #2131: browse is showing internal folders
### 1. What version of Caddy are you using (`caddy -version`)?
`Caddy 0.10.14 (unofficial)` ([abiosoft/caddy-docker](https://github.com/abiosoft/caddy-docker))

### 2. What are you trying to do?
Use browse and internal.

### 3. What is your entire Caddyfile?
```text
        0.0.0.0:80

        root /app

        fastcgi / 127.0.0.1:9000 php
        on startup php-fpm7

        log stdout
        errors stdout


        browse
        internal /bsmenu/.git
        internal /etclist/.git
        internal /haproxy-cf/.git
        internal /nginx-cf/.git
```

### 4. How did you run Caddy (give the full command and describe the execution environment)?
[abiosoft/caddy-docker](https://github.com/abiosoft/caddy-docker)

### 5. Please paste any relevant HTTP request(s) here.

Open the site.

I have the bug visible here https://lib.theel0ja.info/haproxy-cf/

### 6. What did you expect to see?
.git folder (I have it as `internal` in my Caddyfile) should be invisible.

### 7. What did you see instead (give full error messages and/or log)?
[image](https://lib.theel0ja.info/haproxy-cf/)

### 8. How can someone who is starting from scratch reproduce the bug as minimally as possible?

```bash
mkdir should_be_hidden_folder
mkdir other_stuff
```

caddyfile:
```text
:2015

browse
internal /hidden_folder
```

### PR DESCRIPTION
**Pull request by magikstm**
### 1. What does this change do, exactly?
Fixes #2131 
I've not written tests yet as I'm unsure if this fix is the best way to resolve the issue. I reviewed code and I indeed didn't find code in "Internal" that hides these paths.

This commit changes the behavior of the "Internal" middleware and does hide files using it.

### 2. Please link to the relevant issues.
#2131 

### 3. Which documentation changes (if any) need to be made because of this PR?
Add a small mention in https://caddyserver.com/docs/internal to note that internal paths are considered "HiddenFiles" in Caddy and thus not visible in Browse and also not visible directly.

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

**Comment by magikstm** [comment]
@theel0ja which OS are you using? Could you build from source with this commit and confirm if it resolves your issue?

**Comment by theel0ja** [comment]
Using [abiosoft/caddy-docker](https://github.com/abiosoft/caddy-docker) image, that is using Alpine Linux

**Comment by tobya** [comment]
I'll try to test this at the weekend.  Looks straightforward

**Comment by mholt** [comment]
Did anyone test this? @magikstm this approach looks good to me, just write a test to ensure it works if you could. @theel0ja I am not clear on your answer, does this PR solve your issue?

**Comment by magikstm** [comment]
I'm unsure on how to test both "Internal" and "Browse" in a single test in "caddy\caddyhttp\internalsrv\setup_test.go".

**Comment by theel0ja** [comment]
@mholt If it hides folders declared with `internal`, it solves my issue (haven't tried this commit though)

**Comment by mholt** [comment]
@theel0ja When you get a chance, do try it out. :) Then we can be more confident about merging it (without tests...)

**Comment by theel0ja** [comment]
@mholt How can I make binaries for linux/amd64?

**Comment by mholt** [comment]
@theel0ja You need Go installed. Then, on your Linux machine, just run `go get github.com/mholt/caddy/caddy` and it'll be in your `$GOPATH/bin` folder (default GOPATH is `$HOME/go`). Now, for this PR, you'll want to get the code from @magikstm's repository, which you can do either by doing a different `go get` or you can use git remotes to add his repo as a remote and check that out (recommended; see https://github.com/mholt/caddy/blob/master/.github/CONTRIBUTING.md#how-to-make-a-pull-request-to-caddy for instructions). Anyway, `go build` in the folder of the main() will make the binary.

**Comment by linquize** [comment]
@magikstm Could you squash 3 commits into one?

**Comment by magikstm** [comment]
@linquize I could. But, it hasn't been reviewed and may need a test case.

File: caddyhttp/internalsrv/setup.go [hunk 1]
```diff
@@ -33,7 +33,11 @@ func setup(c *caddy.Controller) error {
 		return err
 	}
 
-	httpserver.GetConfig(c).AddMiddleware(func(next httpserver.Handler) httpserver.Handler {
+	//Append Internal paths to Caddy config HiddenFiles
```
Comment (tobya) [inline caddyhttp/internalsrv/setup.go]:
```suggestion
	// Append Internal paths to Caddy config HiddenFiles to ensure
	// files do not appear in Browse
```

**Comment by magikstm** [comment]
Thanks @tobya.

I agree to testing this change in another PR. I'm unsure on how though.

I'm unsure if I could merge my own commits. Please review it again and select "Squash and merge".

**Comment by mholt** [comment]
Thanks for working on this! Let's try it out. Feel free to add more tests any time. :)

**Pull request merged**